includes/Core/Modules/Module.php

```diff
@@ -582,6 +588,38 @@ final public function get_client() {
 		return $this->google_client;
 	}
 
+	/**
+	 * Gets the oAuth client instance to use for the request.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Datapoint $datapoint Datapoint definition.
+	 * @return OAuth_Client OAuth_Client instance.
+	 */
+	private function get_oauth_client_for_request( Datapoint $datapoint ) {
```

Comment:
We may want to rename this method since it gets the OAuth client for a data point, not really the request. I see it's still for the current request of course, but the name seems a bit confusing given the parameter. Alternatively, we could pass the `Data_Request` instance to this method.